revolv/base/views.py

```diff
@@ -59,9 +60,25 @@ def get_context_data(self, **kwargs):
         context["project_dict"] = project_dict
         context["role"] = self.role or "donor"
 
+        context['category_setter_url'] = reverse('dashboard_category_setter')
+        context['categories'] = Category.objects.all().order_by('title')
+        context['preferred_categories'] = RevolvUserProfile.objects.get(user=self.user).preferred_categories.all()
+
         # TODO (noah): add in support for autoshowing a project based on the active_project GET parameter
         return context
 
+class CategoryPreferenceSetterView(UserDataMixin, View):
+    
+    def dispatch(self, request, *args, **kwargs):
+        standard_response = super(CategoryPreferenceSetterView, self).dispatch(request, *args, **kwargs)
+        user = self.user_profile
+        user.preferred_categories.clear()
+        info_dict = request.GET.dict()
+        for category_string in info_dict:
+            if info_dict[category_string] == 'true':
```

Comment:
Yup, we should do this based on id, not title, because titles are not unique. Also, comparing to "true" seems a bit hacky: shouldn't we just pass in an array as a parameter? (To do this, see `getlist` in the docs for [QueryDict](https://docs.djangoproject.com/en/1.7/ref/request-response/))